rfc822/header.go

```diff
@@ -342,14 +342,18 @@ func SetHeaderValue(literal []byte, key, val string) ([]byte, error) {
 	key = textproto.CanonicalMIMEHeaderKey(key)
 	data := joinLine([]byte(key), []byte(val))
 
+	result := make([]byte, 0, len(data)+len(literal))
 	if !foundFirstEntry {
-		return append(rawHeader, append(data, body...)...), nil
+		result = append(result, rawHeader...)
+		result = append(result, data...)
+		result = append(result, body...)
```

Comment:
ahhh good catch

seeing as append caused problems in the past... maybe you wanna make it even safer, like this?

```suggestion
		result = bytes.Join([][]byte{rawHeader, data, body}, nil)
```

Reply:
`result` is a new value, it's practically the same as that statement.

Reply:
I have also updated the tests to check for this.

Reply:
alright, i'm just paranoid about appends now, but nevermind